src/jit/optcse.cpp

```diff
@@ -98,7 +102,7 @@ void Compiler::optUnmarkCSE(GenTree* tree)
     // make sure it's been initialized
     noway_assert(optCSEweight <= BB_MAX_WEIGHT);
 
-    /* Is this a CSE use? */
+    // Is this a CSE use?
     if (IS_CSE_USE(tree->gtCSEnum))
```

Comment:
Isn't this just `IS_CSE_USE(CSEnum)` ?

Reply:
Well not exactly,  GET_CSE_INDEX returns the absolute value of the gtCSEnum, disacrding the sign bit.
IS_CSE_USE is checking the sign bit of gtCSEnum, if is is positive it is a use and if it is negative it is a def.

```
#define IS_CSE_INDEX(x) (x != 0)
#define IS_CSE_USE(x) (x > 0)
#define IS_CSE_DEF(x) (x < 0)
#define GET_CSE_INDEX(x) ((x > 0) ? x : -x)
```

Reply:
Ah, ok. Well maybe sink `CSEnum` down under the `if` then too, since you don't look at it unless you have a use.